File: go/isogram/isogram.go
// Package isogram has the function IsIsogram which tests if a given word is an isogram
package isogram

import "unicode"

// IsIsogram checks if a given word is an isogram
func IsIsogram(word string) bool {
	// For having a record of all visited letters of the word
	prevLetters := make(map[rune]bool)

	for _, letter := range word {
		if letter == ' ' || letter == '-' {
			continue
		}

		letter = unicode.ToLower(letter)
		if prevLetters[letter] {
			return false
		}

		prevLetters[letter] = true
	}

	return true
}

File: python/isogram/isogram.py
def is_isogram(string: str) -> bool:
    """Checks if a given string is an Isogram (ie. no repeated chars)
    Parameters:
        string the given string
    Retunrs:
    ========
        True, if string is an isogram
        False, otherwise
    """

    # Controls whether a char has already appeared or not
    char_has_appeared = set()

    # Cleans the input
    cleaned = string
    cleaned = cleaned.replace(" ", "")
    cleaned = cleaned.replace("-", "")
    cleaned = cleaned.lower()

    for char in cleaned:
        # We have found a repeated lchar
        if char in char_has_appeared:
            return False

        # We mark the char as already visited
        char_has_appeared.add(char)

    return True
